.github/workflows/smoke-tests.yml

```
@@ -34,7 +34,7 @@ jobs:
       - name: 'preparation: Checkout MTL'
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
-          ref: 'pstaszczuk/smoke-tests'
+          ref: '${{ github.ref }}'
```

This could be made like this probably - the bellow would be the 1st line:
```yaml
on:
  workflow_dispatch:
    inputs:
      branch-to-checkout:
        type: string
        default: 'main'
        required: false
        description: 'Branch name to use'
  push:
    branches:
      - main
      - 'maint-**'
      - 'v*'
  pull_request:
    branches:
      - main
      - 'maint-**'
      - 'v*'

concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
  cancel-in-progress: true
```

And then this suggestion to close the ref choosing

```suggestion
          ref: '${{ inputs.branch-to-checkout || github.sha || github.ref }}''
```